### PR TITLE
Update CVM attestation sample app and client library to enable self-contained attestation client.

### PR DESCRIPTION
**Pull request by deepaksh-microsoft**
Customers can now build, install and use the attestation client lib along with the attestation client app on Linux Confidential/Trusted virtual machines, instead of only relying on pre-released lib package.

### CODE REVIEW

**Review by mishra-k — APPROVED** [review]
LGTM!

File: client-library/src/Attestation/LinuxTpm/tools/cmake/FindGTest.cmake [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Try to find GTest
-find_path(GTEST_INCLUDE_DIR NAMES gtest PATHS ${SYSTEM_LIBRARY_PREFIX}/usr/local/include ${SYSTEM_LIBRARY_PREFIX}/usr/include NO_DEFAULT_PATH)
```
Comment (Bab95) [inline client-library/src/Attestation/LinuxTpm/tools/cmake/FindGTest.cmake:2]: we shouldn't remove NO_DEFAULT_PATH flag from here.

File: client-library/src/Attestation/pre-requisites.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    rpm \
+    debhelper \
+    libgtest-dev \
+    libgmock-dev \
+    cmake \
+    git \
+    zip \
+    uuid-dev \
+    libjson-c-dev \
+	libarchive-dev \
```
Comment (Bab95) [inline client-library/src/Attestation/pre-requisites.sh:36]: [nit] spacing

File: client-library/src/Attestation/pre-requisites.sh [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    sudo rm -rf openssl-3.3.2.tar.gz && \
+    cd /tmp/openssl-3.3.2 && \
+    sudo ./config --prefix=/usr/local/attestationssl --openssldir=/usr/local/attestationssl && \
+    sudo make -j$(nproc) && \
+    sudo make install_sw
+
+cd ${CURRENT_DIR}
+
+export LDFLAGS="-L/usr/local/attestationssl/lib64 $LDFLAGS" && \ 
+	export CPPFLAGS="-I/usr/local/attestationssl/include $CPPFLAGS" && \
```
Comment (Bab95) [inline client-library/src/Attestation/pre-requisites.sh:58]: [nit] spacing

File: client-library/src/Attestation/LinuxTpm/tools/cmake/FindGTest.cmake (hunk 1 above)
Comment (deepaksh-microsoft) [reply client-library/src/Attestation/LinuxTpm/tools/cmake/FindGTest.cmake:2]: gtest and gmock both installed in pre-requisites at default path.

File: client-library/src/Attestation/pre-requisites.sh [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    sudo make -j$(nproc) && \
+    sudo make LIBDIR=lib && sudo make install
+
+cd ${CURRENT_DIR}
+
+sudo mkdir -p /usr/src
+
+export CC=gcc && export CXX=g++ && \
+    export LDFLAGS="-L/usr/local/attestationssl/lib64 -lcrypto $LDFLAGS" && \
+    export CPPFLAGS="-I/usr/local/attestationssl/include $CPPFLAGS" && \
```
[the remark this answers is not in the document]
Comment (deepaksh-microsoft) [reply client-library/src/Attestation/pre-requisites.sh:72]: updated using pkg_config_path. Also, changed the installation location of tpm2-tss as it is attestation usage specific and doesn't direct to default dependency directories.

File: client-library/src/Attestation/pre-requisites.sh [hunk 5]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+sudo apt-get update
```
[the remark this answers is not in the document]
Comment (deepaksh-microsoft) [reply client-library/src/Attestation/pre-requisites.sh:3]: The existing instructions are also only for canonical based images. As other distros start supporting CVMs, we can start adding other flavors of scripts.

**Review by Bab95 — APPROVED** [review]
looks good to me.

File: client-library/src/Attestation/AttestationClient/lib/DynamicLibrary/CMakeLists.txt [hunk 6]
```diff
@@ -42,7 +45,7 @@ add_library(${CMAKE_PROJECT_TARGET} SHARED ../../AttestationHelper.cpp
                                            ../AttestationLibTelemetry.cpp
                                            ${CMAKE_SOURCE_DIR}/../external/jsoncpp-0.10.7/src/jsoncpp.cpp)
                                            
-find_library(CURL_LIB NAMES libcurl.a PATHS /usr/lib
+find_library(CURL_LIB NAMES libcurl.a PATHS /usr/local/attestationcurl/lib
```
Comment (Bab95) [inline client-library/src/Attestation/AttestationClient/lib/DynamicLibrary/CMakeLists.txt]: should this also be changed to dynamic?

File: client-library/src/Attestation/AttestationClient/tests/lib/CMakeLists.txt [hunk 7]
```diff
@@ -54,10 +54,10 @@ add_executable(${CMAKE_PROJECT_TARGET} ClientLibTests.cpp
                                        ../../lib/AttestationLibTelemetry.cpp
                                        ${CMAKE_SOURCE_DIR}/../external/jsoncpp-0.10.7/src/jsoncpp.cpp)
 
-find_library(CURL_LIB NAMES libcurl.a PATHS /usr/lib
+find_library(CURL_LIB NAMES libcurl.a PATHS /usr/local/attestationcurl/lib
```
[the remark this answers is not in the document]
Comment (Bab95) [reply client-library/src/Attestation/AttestationClient/tests/lib/CMakeLists.txt]: could be ignored for tests

**Review by Bab95 — APPROVED** [review]

**Pull request merged**